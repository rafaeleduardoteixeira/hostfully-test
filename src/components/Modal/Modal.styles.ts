import styled from 'styled-components';
import { colors, media } from '../../themes';

export const Container = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  z-index: 100;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.5);

  display: flex;
  justify-content: center;
  align-items: center;
`;

export const ModalContent = styled.div<{ width?: string }>`
  display: flex;
  background-color: white;
  padding: 45px 20px 20px 20px;
  border-radius: 10px;
  width: ${({ width }) => width || '60%'};
  overflow: auto;
  max-height: 600px;
  position: relative;

  @media ${media.tablet} {
    width: 90%;
  }

  @media ${media.mobile} {
    width: 80%;
    max-height: 60%;
  }
`;

export const CloseButton = styled.button`
  position: absolute;
  top: 10px;
  right: 15px;
  background-color: transparent;
  border: none;
  cursor: pointer;
  font-size: 24px;

  width: 20px;
  height: 20px;

  &:before {
    color: ${colors.black};
    content: 'Ã—';
  }
`;
