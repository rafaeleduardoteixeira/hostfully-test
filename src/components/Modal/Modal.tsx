import React, { ReactElement } from 'react';
import { CloseButton, Container, ModalContent } from './Modal.styles';

interface ModalProps {
  width?: string;
  children: ReactElement;
  onClose?: () => void;
}

export const Modal = ({ children, width, onClose }: ModalProps) => {
  return (
    <Container data-testid="Modal">
      <ModalContent width={width}>
        {onClose && <CloseButton onClick={onClose} />}
        {children}
      </ModalContent>
    </Container>
  );
};
