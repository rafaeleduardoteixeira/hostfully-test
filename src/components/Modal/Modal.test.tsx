import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { Modal } from './Modal';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe('Modal', () => {
  test('Snapshot test', () => {
    const { asFragment } = render(
      <Modal>
        <div>Modal</div>
      </Modal>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('Render', () => {
    render(
      <Modal onClose={jest.fn()}>
        <div>Modal</div>
      </Modal>,
    );
    expect(screen.getByText('Modal')).toBeInTheDocument();
  });
});
