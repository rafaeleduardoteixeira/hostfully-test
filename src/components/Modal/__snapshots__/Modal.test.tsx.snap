// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="sc-braxZu fTvrrx"
    data-testid="Modal"
  >
    <div
      class="sc-gJhJTp kTJVVP"
    >
      <div>
        Modal
      </div>
    </div>
  </div>
</DocumentFragment>
`;
